dispatcher.go

```diff
@@ -103,16 +111,35 @@ func NewDispatcher(cfg Config) *Dispatcher {
 		panic("yarpc.NewDispatcher expects a valid service name: %s" + err.Error())
 	}
 
+	logger := zap.NewNop()
+	if cfg.logger != nil {
+		logger = cfg.logger.Named("yarpc").With(
+			zap.Namespace("yarpc"), // isolate yarpc's keys
+			zap.String("dispatcher", cfg.Name),
```

Comment:
The namespace applies to all subsequent keys. This has the effect of producing this unfinished json blob:

```
{"yarpc": {"dispatcher": "foo",
```

which we'll then keep appending to.